### PR TITLE
provide one link to run all failed tests

### PR DESCRIPTION
**Pull request by OscarQQ**
Signed-off-by: OscarQQ <Yixin.Qian@ibm.com>

### CODE REVIEW

**Comment by OscarQQ** [comment]
Grinder_TKG/799 only one failed test, nothing changed
Grinder_TKG/800 two (and more) failed tests, the link with (target=testList TESTLIST) will show

example console output:
```
01:04:10  ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
01:04:10  To rebuild the failed test in a jenkins job, copy the following link and fill out the <Jenkins URL> and <FAILED test target>:
01:04:10  <Jenkins URL>/parambuild/?JDK_VERSION=8&JDK_IMPL=openj9&BUILD_LIST=functional/TestExample&PLATFORM=x86-64_linux&TARGET=<FAILED test target>
01:04:10  
01:04:10  For example, to rebuild the failed tests in <Jenkins URL>=https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder, use the following links:
01:04:10  https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/parambuild/?JDK_VERSION=8&JDK_IMPL=openj9&BUILD_LIST=functional/TestExample&PLATFORM=x86-64_linux&TARGET=testExample1_0
01:04:10  https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/parambuild/?JDK_VERSION=8&JDK_IMPL=openj9&BUILD_LIST=functional/TestExample&PLATFORM=x86-64_linux&TARGET=testExample1_1
01:04:10  rebuild the failed tests in one link
01:04:10  https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/parambuild/?JDK_VERSION=8&JDK_IMPL=openj9&BUILD_LIST=functional/TestExample&PLATFORM=x86-64_linux&TARGET=testList%20TESTLIST=testExample1_0,testExample1_1
01:04:10  ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
```

**Comment by OscarQQ** [comment]
@renfeiw @llxia could you review this? thanks

File: scripts/resultsSum.pl [hunk 1]
```diff
@@ -241,8 +242,14 @@ sub resultReporter {
 		print "<Jenkins URL>/parambuild/?JDK_VERSION=$jdkVersion&JDK_IMPL=$jdkImpl$buildParam$platformParam$customTargetParam&TARGET=<FAILED test target>\n\n";
 		print "For example, to rebuild the failed tests in <Jenkins URL>=${hudsonUrl}job/Grinder, use the following links:\n";
 		foreach my $failedTarget (@failed) {
+			$failedList = $failedList . ${failedTarget} . ",";
 			print "${hudsonUrl}job/Grinder/parambuild/?JDK_VERSION=$jdkVersion&JDK_IMPL=$jdkImpl$buildParam$platformParam$customTargetParam&TARGET=$failedTarget\n";
 		}
+		if ($numOfFailed > 1) {
+			chop($failedList);
+			print "rebuild the failed tests in one link\n";
```
Comment (llxia) [inline scripts/resultsSum.pl:250]: Please add `:` at the end

File: scripts/resultsSum.pl [hunk 2]
```diff
@@ -241,8 +242,14 @@ sub resultReporter {
...
+			print "${hudsonUrl}job/Grinder/parambuild/?JDK_VERSION=$jdkVersion&JDK_IMPL=$jdkImpl$buildParam$platformParam$customTargetParam&TARGET=testList%20TESTLIST=$failedList\n";
```
Comment (llxia) [inline scripts/resultsSum.pl:251]: Could we make this a parameter? `parambuild/?JDK_VERSION=$jdkVersion&JDK_IMPL=$jdkImpl$buildParam$platformParam$customTargetParam`

It is used in Line 242 and 246.

File: scripts/resultsSum.pl [hunk 3]
```diff
@@ -241,8 +242,14 @@ sub resultReporter {
 		print "<Jenkins URL>/parambuild/?JDK_VERSION=$jdkVersion&JDK_IMPL=$jdkImpl$buildParam$platformParam$customTargetParam&TARGET=<FAILED test target>\n\n";
 		print "For example, to rebuild the failed tests in <Jenkins URL>=${hudsonUrl}job/Grinder, use the following links:\n";
 		foreach my $failedTarget (@failed) {
+			$failedList = $failedList . ${failedTarget} . ",";
 			print "${hudsonUrl}job/Grinder/parambuild/?JDK_VERSION=$jdkVersion&JDK_IMPL=$jdkImpl$buildParam$platformParam$customTargetParam&TARGET=$failedTarget\n";
 		}
+		if ($numOfFailed > 1) {
+			chop($failedList);
```
Comment (renfeiw) [inline scripts/resultsSum.pl:249]: Maybe it's more concise to use `join(",", @failed)` than creating a `$faliledList`.

**Comment by OscarQQ** [comment]
Grinder_TKG/802

**Review by renfeiw — APPROVED** [review]

**Comment by OscarQQ** [comment]
/Grinder_TKG/804

**Review by llxia — APPROVED** [review]
LGTM

**Pull request merged**